jobs/geo/core.py

```diff
@@ -1,44 +1,90 @@
 from __future__ import annotations
 
-from io import StringIO
-
 from jobs.utils.snowflake import gdf_to_snowflake, snowflake_connection_from_environment
 
 
-def load_geo_data(conn, url: str, table: str) -> None:
-    """Load general geospatial data. Given a URL, load geospatial data into Snowflake."""
-    import geopandas
-    import requests
+def load_ca_geo_data(conn, year: str) -> None:
+    """Load CA Census geo data into Snowflake."""
+    from pygris import (
+        block_groups,
+        blocks,
+        counties,
+        county_subdivisions,
+        places,
+        primary_secondary_roads,
+        pumas,
+        tracts,
+    )
+
+    print(f"Downloading data for CA in year {year}")
+
+    state = "CA"
+    cache = True
+    dfs = []
+
+    county = counties(state=state, year=year, cache=cache)
+    county["year"] = year
+    dfs.append(county)
+
+    tract = tracts(state=state, year=year, cache=cache)
+    tract["year"] = year
+    dfs.append(tract)
 
-    print(f"Downloading data {table} from {url}")
+    block_group = block_groups(state=state, year=year, cache=cache)
+    block_group["year"] = year
+    dfs.append(block_group)
 
-    # Some of the target sources come from Esri servers, and GDAL's vsi system
-    # can be cranky about reading from them (issues with range requests and other
-    # weird errors). So instead we load the data first into a text blob using
-    # requests, then hand it off to geopandas.
-    f = StringIO(requests.get(url).text)
-    gdf = geopandas.read_file(f)
+    block = blocks(state=state, year=year, cache=cache)
+    block["year"] = year
+    dfs.append(block)
 
-    print(f"Writing {table} to snowflake")
-    gdf_to_snowflake(
-        gdf,
-        conn,
-        table_name=table,
-        cluster=False,
+    place = places(state=state, year=year, cache=cache)
+    place["year"] = year
+    dfs.append(place)
+
+    puma = pumas(state=state, year=int(year), cache=cache)
```

Comment:
Why do you need to convert `year` to an `int` here, but not elsewhere?

Reply:
I don't recall the exact errors, but basically when running the script it would error out and say can't compare string to int. I thought it was weird it only did that for some but not all as well. When I tested this in jupyter lab it worked without the conversion.

Reply:
done